### PR TITLE
feat: Adding support for AWS Secrets Manager

### PR DESCRIPTION
**Pull request by jvasallo**
Adding support for using AWS Secrets Manager.

### CODE REVIEW

File: src/foremast/templates/infrastructure/iam/kms.json.j2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+        {
+            "Sid": "KMSDecrypt",
+            "Action": [
+                "kms:Decrypt"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+            {%- for key_id in items %}
```
Comment (sijis) [inline src/foremast/templates/infrastructure/iam/kms.json.j2]: Where is `items` coming from?

**Comment by sijis** [comment]
How are some if these values, namely `items`, sent to these templates?

The templates do not seem to be included in any of the other ones.

Would adding these templates to your internal template files suffice?

File: src/foremast/templates/infrastructure/iam/kms.json.j2 (hunk 1 above)
Comment (jvasallo) [reply src/foremast/templates/infrastructure/iam/kms.json.j2]: Relying on the internal services code that creates items:
https://github.com/foremast/foremast/blob/432284ae32b7f30df99d1bb5eae0a393f9154dd1/src/foremast/iam/construct_policy.py#L111

Adding to our internal templates is fine, but just wanted to release a way of doing it to the community. :)

Comment (jvasallo) [reply src/foremast/templates/infrastructure/iam/kms.json.j2]: A good example of it used today is: https://github.com/foremast/foremast/blob/master/src/foremast/templates/infrastructure/iam/dynamodb.json.j2

**Pull request merged**